## Loading Dataset and Initial Inspection


### Import & Load Data

In [114]:
import pandas as pd
import numpy as np

df=pd.read_csv("data/call_center_data.csv")

### Initial Inspection


In [115]:
print("Shape:", df.shape)

df.info()

df.head()

Shape: (968, 16)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 968 entries, 0 to 967
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   call_id                   968 non-null    object 
 1   call_date                 968 non-null    object 
 2   start_time                968 non-null    object 
 3   customer_phone            968 non-null    object 
 4   call_reason               968 non-null    object 
 5   agent_id                  964 non-null    object 
 6   queue                     968 non-null    object 
 7   channel                   968 non-null    object 
 8   status                    968 non-null    object 
 9   wait_seconds              968 non-null    int64  
 10  handle_seconds            848 non-null    float64
 11  hold_seconds              846 non-null    float64
 12  after_call_work_seconds   846 non-null    float64
 13  csat_score                846 non-null    float6

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count
0,C0001,1/1/2026,18:46:03,912001304,Account Access,A06,Billing,Phone,Answered,84,187.0,24.0,62.0,3.0,No,0
1,C0002,1/1/2026,15:07:48,912000398,Order Status,A07,Sales,Phone,Abandoned,88,NaN,NaN,NaN,NaN,NaN,0
2,C0003,1/1/2026,16:41:35,912000387,Account Access,A12,Sales,Callback,Answered,39,219.0,48.0,42.0,4.0,No,0
3,C0004,1/1/2026,14:09:47,912000550,Technical Problem,A12,Support,Phone,Answered,141,396.0,48.0,30.0,4.0,Yes,0
4,C0005,1/1/2026,15:05:56,912000598,Product Inquiry,A10,Sales,Phone,Answered,33,253.0,59.0,51.0,3.0,Yes,0


### Missing Values & Duplicates

In [116]:
print("Missing values:")
print(df.isnull().sum())

print("\nExact duplicate rows:", df.duplicated().sum())

Missing values:
call_id                       0
call_date                     0
start_time                    0
customer_phone                0
call_reason                   0
agent_id                      4
queue                         0
channel                       0
status                        0
wait_seconds                  0
handle_seconds              120
hold_seconds                122
after_call_work_seconds     122
csat_score                  122
first_contact_resolution    122
transfer_count                0
dtype: int64

Exact duplicate rows: 2


### Descriptive Statistics

In [117]:
df.describe(include="all")

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count
count,968,968,968,968,968,964,968,968,968,968.000000,848.000000,846.000000,846.000000,846.000000,846,968.000000
unique,965,29,950,966,8,12,3,2,3,NaN,NaN,NaN,NaN,NaN,2,NaN
top,C0500,1/10/2026,16:06:45,912000934,Technical Problem,A11,Support,Phone,Answered,NaN,NaN,NaN,NaN,NaN,Yes,NaN
freq,2,70,2,2,208,94,448,796,740,NaN,NaN,NaN,NaN,NaN,612,NaN
mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,98.318182,306.278302,42.968085,47.784870,3.591017,NaN,0.373967
std,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,52.009426,110.535139,21.625651,15.830868,1.174931,NaN,0.632312
min,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,10.000000,-120.000000,5.000000,20.000000,1.000000,NaN,0.000000
25%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,56.000000,218.000000,25.000000,35.000000,3.000000,NaN,0.000000
50%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,88.000000,284.000000,42.000000,48.000000,4.000000,NaN,0.000000
75%,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,139.000000,390.000000,62.000000,62.000000,5.000000,NaN,1.000000


# Data Cleaning

### Categorical & Numeric Inspection

In [118]:
categorical_cols = [
    "call_reason",  
    "queue",
    "channel",
    "status",
    "first_contact_resolution"
]

for col in categorical_cols:
    print(f"\n--- {col} ---")
    print(df[col].value_counts(dropna=False))


--- call_reason ---
call_reason
Technical Problem      208
Account Access         157
Product Inquiry        125
Order Status           115
Billing Issue          108
Complaint               99
Subscription Change     79
Refund Request          77
Name: count, dtype: int64

--- queue ---
queue
Support    448
Sales      290
Billing    230
Name: count, dtype: int64

--- channel ---
channel
Phone       796
Callback    172
Name: count, dtype: int64

--- status ---
status
Answered       740
Abandoned      122
Transferred    106
Name: count, dtype: int64

--- first_contact_resolution ---
first_contact_resolution
Yes    612
No     234
NaN    122
Name: count, dtype: int64


In [119]:
numeric_cols = [
    "wait_seconds",
    "handle_seconds",
    "hold_seconds",
    "after_call_work_seconds",
    "csat_score",
    "transfer_count"
]

for col in numeric_cols:
    print(f"\n--- {col} ---")
    print(df[col].describe())


--- wait_seconds ---
count    968.000000
mean      98.318182
std       52.009426
min       10.000000
25%       56.000000
50%       88.000000
75%      139.000000
max      257.000000
Name: wait_seconds, dtype: float64

--- handle_seconds ---
count    848.000000
mean     306.278302
std      110.535139
min     -120.000000
25%      218.000000
50%      284.000000
75%      390.000000
max      663.000000
Name: handle_seconds, dtype: float64

--- hold_seconds ---
count    846.000000
mean      42.968085
std       21.625651
min        5.000000
25%       25.000000
50%       42.000000
75%       62.000000
max       80.000000
Name: hold_seconds, dtype: float64

--- after_call_work_seconds ---
count    846.000000
mean      47.784870
std       15.830868
min       20.000000
25%       35.000000
50%       48.000000
75%       62.000000
max       75.000000
Name: after_call_work_seconds, dtype: float64

--- csat_score ---
count    846.000000
mean       3.591017
std        1.174931
min        1.000000
25%   

### Negative & Zero Values

In [120]:
for col in [
    "wait_seconds",
    "handle_seconds",
    "hold_seconds",
    "after_call_work_seconds",
]:
    print(f"\n--- {col} ---")
    print("Negative values:", (df[col] < 0).sum())
    print("Zero values:", (df[col] == 0).sum())


--- wait_seconds ---
Negative values: 0
Zero values: 0

--- handle_seconds ---
Negative values: 2
Zero values: 0

--- hold_seconds ---
Negative values: 0
Zero values: 0

--- after_call_work_seconds ---
Negative values: 0
Zero values: 0


### CSAT Values

In [121]:
print(df["csat_score"].value_counts(dropna=False).sort_index())

csat_score
1.0     41
2.0    126
3.0    200
4.0    253
5.0    223
6.0      3
NaN    122
Name: count, dtype: int64


### Inspect CSAT = 6

In [122]:
df[df["csat_score"] == 6]

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count
452,C0453,1/13/2026,18:07:24,912001079,Account Access,A08,Support,Callback,Answered,89,422.0,43.0,65.0,6.0,Yes,0
847,C0851,1/26/2026,9:51:33,912000400,Product Inquiry,A06,Support,Phone,Answered,179,431.0,40.0,71.0,6.0,Yes,1
934,C0938,1/29/2026,9:11:32,912000476,Account Access,A02,Billing,Phone,Answered,46,175.0,67.0,67.0,6.0,Yes,0


### CSAT Missingness by Status

In [123]:
pd.crosstab(
    df["status"],
    df["csat_score"].isna()
)

csat_score,False,True
status,,
Abandoned,0,122
Answered,740,0
Transferred,106,0


### Create Clean Handle Time

In [124]:
df["handle_seconds_clean"] = df["handle_seconds"].where(
    df["handle_seconds"] >= 0,
    np.nan
)

### Investigate Duplicate Call IDs

In [125]:
df[df.duplicated("call_id", keep=False)].sort_values("call_id")

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count,handle_seconds_clean
49,C0050,1/2/2026,15:05:25,912000934,Technical Problem,A04,Support,Callback,Answered,87,352.0,68.0,58.0,3.0,Yes,1,352.0
965,C0050,1/2/2026,15:05:25,912000934,Technical Problem,A04,Support,Callback,Answered,87,352.0,68.0,58.0,3.0,Yes,1,352.0
199,C0200,1/7/2026,19:29:46,912001300,Product Inquiry,A07,Support,Phone,Transferred,107,652.0,50.0,68.0,5.0,No,0,652.0
966,C0200,1/7/2026,19:29:46,912001300,Product Inquiry,A07,Support,Phone,Transferred,107,652.0,50.0,68.0,5.0,No,0,652.0
499,C0500,1/15/2026,16:29:48,912000206,Subscription Change,A05,Support,Phone,Answered,173,327.0,32.0,31.0,2.0,No,0,327.0
967,C0500,1/15/2026,16:29:48,+98912000206,Subscription Change,A05,Support,Phone,Answered,173,327.0,32.0,31.0,2.0,No,0,327.0


### Remove Exact Duplicates

In [126]:
df = df.drop_duplicates()
print("Shape after removing exact duplicates:", df.shape)
print("Exact duplicates:", df.duplicated().sum())


Shape after removing exact duplicates: (966, 17)
Exact duplicates: 0


### Investigate inconsistent duplicate


In [127]:
df[df["customer_phone"].astype(str).str.startswith("+98")]

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count,handle_seconds_clean
272,C0273,1/9/2026,14:50:14,"+98912001153, +98912004453",Billing Issue,NaN,Support,Phone,Answered,115,412.0,73.0,20.0,2.0,Yes,0,412.0
451,C0452,1/13/2026,15:31:50,+98912001460,Technical Problem,A08,Support,Phone,Answered,130,367.0,74.0,54.0,5.0,No,0,367.0
928,C0932,1/28/2026,17:29:28,+98912000468,Technical Problem,A03,Billing,Callback,Answered,44,242.0,78.0,41.0,5.0,No,1,242.0
933,C0937,1/29/2026,14:53:24,+98912001498,Billing Issue,A12,Sales,Callback,Answered,37,252.0,56.0,32.0,5.0,Yes,0,252.0
967,C0500,1/15/2026,16:29:48,+98912000206,Subscription Change,A05,Support,Phone,Answered,173,327.0,32.0,31.0,2.0,No,0,327.0


### Remove the inconsistent duplicate

In [128]:
df = df[df["call_id"] != "C0500"]

print("Shape after removing inconsistent duplicate:", df.shape)

Shape after removing inconsistent duplicate: (964, 17)


### Validate Duplicate Removal

In [129]:
print("Unique Call IDs:", df["call_id"].nunique())
print("Exact Duplicates:", df.duplicated().sum())

Unique Call IDs: 964
Exact Duplicates: 0


### Missing Agent IDs

In [130]:
df[df["agent_id"].isna()]

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count,handle_seconds_clean
16,C0017,1/1/2026,8:40:41,912000706,Refund Request,NaN,Support,Phone,Answered,106,344.0,24.0,41.0,5.0,Yes,0,344.0
195,C0196,1/7/2026,18:24:33,0912000870,Product Inquiry,NaN,Sales,Phone,Answered,42,244.0,77.0,49.0,3.0,No,0,244.0
272,C0273,1/9/2026,14:50:14,"+98912001153, +98912004453",Billing Issue,NaN,Support,Phone,Answered,115,412.0,73.0,20.0,2.0,Yes,0,412.0
863,C0867,1/26/2026,19:37:24,912000212,Order Status,NaN,Sales,Phone,Answered,86,213.0,31.0,41.0,5.0,Yes,0,213.0


### Status vs Handle Time

In [131]:
pd.crosstab(
    df["status"],
    df["handle_seconds_clean"].notna())

handle_seconds_clean,False,True
status,,
Abandoned,120,2
Answered,2,735
Transferred,0,105


### Find Logical Anomalies


In [132]:
df[
    ((df["status"] == "Abandoned") & df["handle_seconds_clean"].notna()) |
    ((df["status"] == "Answered") & df["handle_seconds_clean"].isna())
]

,call_id,call_date,start_time,customer_phone,call_reason,agent_id,queue,channel,status,wait_seconds,handle_seconds,hold_seconds,after_call_work_seconds,csat_score,first_contact_resolution,transfer_count,handle_seconds_clean
15,C0016,1/1/2026,15:10:23,912000445,Account Access,A07,Support,Phone,Abandoned,185,40.0,NaN,NaN,NaN,NaN,0,40.0
146,C0147,1/5/2026,17:46:57,912000289,Technical Problem,A03,Support,Phone,Answered,132,-45.0,74.0,42.0,1.0,No,2,NaN
490,C0491,1/15/2026,18:06:53,912000166,Complaint,A08,Support,Phone,Abandoned,179,25.0,NaN,NaN,NaN,NaN,0,25.0
554,C0555,1/17/2026,11:40:27,912000657,Product Inquiry,A09,Sales,Phone,Answered,64,-120.0,56.0,26.0,5.0,Yes,0,NaN


### Exclude Handle Time for Abandoned Calls

In [133]:
df.loc[
    df["status"] == "Abandoned",
     "handle_seconds_clean"
     ]= np.nan

### Hold vs Handle Consistency

In [134]:
(df["hold_seconds"] > df["handle_seconds_clean"]).sum()

np.int64(0)

### Convert Date & Time

In [135]:
df["call_date"] = pd.to_datetime(df["call_date"])

df["start_time"] = pd.to_datetime(
    df["start_time"],
    format="%H:%M:%S"
)

### Validate Conversion

In [136]:

print(df["call_date"].isna().sum())
print(df["start_time"].isna().sum())

0
0


### Flag potentially unusual CSAT values

In [137]:
df["csat_invalid"] = df["csat_score"] > 5

print("Potentially unusual CSAT values:")
print(
    df.loc[
        df["csat_invalid"],
        ["call_id", "csat_score"]
    ]
)

Potentially unusual CSAT values:
    call_id  csat_score
452   C0453         6.0
847   C0851         6.0
934   C0938         6.0


CSAT values above 5 were flagged for investigation. Since the dataset documentation does not define a strict CSAT scale, these values were retained in the analysis.

### Extract Hour

In [138]:
df["hour"] = df["start_time"].dt.hour

df["hour"].value_counts().sort_index()

hour
8      56
9      72
10     81
11     86
12     49
14    114
15    129
16    127
17    112
18     79
19     59
Name: count, dtype: int64

### Define Shifts

In [139]:
conditions = [
    df["hour"].between(8, 12),
    df["hour"].between(14, 19)
]

choices = ["Morning", "Evening"]

df["shift"] = np.select(conditions, choices, default="Unknown")

### Validate Shifts

In [140]:
df["shift"].value_counts()

shift
Evening    620
Morning    344
Name: count, dtype: int64

### Save Cleaned Dataset

In [141]:
df.to_csv(
    "data/call_center_cleaned.csv",
    index=False
)

print("Cleaned dataset saved successfully.")

Cleaned dataset saved successfully.


# KPI Analysis

### Call counts by status

In [142]:
status_counts = df["status"].value_counts()
total_calls = len(df)
answered_calls = (df["status"] == "Answered").sum()
transferred_calls = (df["status"] == "Transferred").sum()
abandoned_calls = (df["status"] == "Abandoned").sum()

print("Total Calls:", total_calls)
print("Answered:", answered_calls)
print("Transferred:", transferred_calls)
print("Abandoned:", abandoned_calls)

Total Calls: 964
Answered: 737
Transferred: 105
Abandoned: 122


### Answer Rate

In [143]:
handled_calls = answered_calls + transferred_calls

answer_rate = (handled_calls / total_calls) * 100

print("Answer Rate:", round(answer_rate, 2), "%")

Answer Rate: 87.34 %


### Abandon Rate

In [144]:
total_calls = len(df)
abandoned_calls = (df["status"] == "Abandoned").sum()

abandon_rate = (abandoned_calls / total_calls) * 100

print("Total Calls:", total_calls)
print("Abandoned Calls:", abandoned_calls)
print("Abandon Rate:", round(abandon_rate, 2), "%")

Total Calls: 964
Abandoned Calls: 122
Abandon Rate: 12.66 %


### Average Wait Time (AWT)


In [145]:
awt = df["wait_seconds"].mean()

print("Average Wait Time (AWT):", round(awt, 2), "seconds")

Average Wait Time (AWT): 98.17 seconds


### Average Handle Time (AHT)

In [146]:
aht = df["handle_seconds_clean"].mean()

print("Average Handle Time (AHT):", round(aht, 2), "seconds")

Average Handle Time (AHT): 307.34 seconds


### Customer Satisfaction (CSAT)

In [147]:

csat_mean = df["csat_score"].mean()

print("Average CSAT:", round(csat_mean, 2))

Average CSAT: 3.59


### FCR distribution

In [148]:
df["first_contact_resolution"].value_counts(dropna=False)

first_contact_resolution
Yes    611
No     231
NaN    122
Name: count, dtype: int64

### First Contact Resolution (FCR)

In [149]:
fcr_valid = df["first_contact_resolution"].dropna()

fcr = (fcr_valid == "Yes").mean() * 100

print("FCR:", round(fcr, 2), "%")

FCR: 72.57 %


### Transfer Rate


In [150]:
transferred_calls = (df["status"] == "Transferred").sum()
total_calls = len(df)

transfer_rate = (transferred_calls / total_calls) * 100

print("Transfer Rate:", round(transfer_rate, 2), "%")

Transfer Rate: 10.89 %


### Average After Call Work (ACW)

In [151]:
acw = df["after_call_work_seconds"].mean()

print("Average ACW:", round(acw, 2), "seconds")

Average ACW: 47.79 seconds


### Contact volume by hour

In [152]:
hourly_volume = df["hour"].value_counts().sort_index()

print(hourly_volume)

hour
8      56
9      72
10     81
11     86
12     49
14    114
15    129
16    127
17    112
18     79
19     59
Name: count, dtype: int64


### Peak Hour

In [153]:
peak_hour = hourly_volume.idxmax()
peak_volume = hourly_volume.max()

print("Peak Hour:", peak_hour)
print("Number of Calls:", peak_volume)

Peak Hour: 15
Number of Calls: 129


### Contact volume by day

In [154]:
daily_volume = df["call_date"].value_counts().sort_index()

print(daily_volume)

call_date
2026-01-01    33
2026-01-02    33
2026-01-03    32
2026-01-04    32
2026-01-05    32
2026-01-06    32
2026-01-07    32
2026-01-08    32
2026-01-09    32
2026-01-10    70
2026-01-11    32
2026-01-12    32
2026-01-13    32
2026-01-14    32
2026-01-15    31
2026-01-16    32
2026-01-17    32
2026-01-18    32
2026-01-19    32
2026-01-20    32
2026-01-21    32
2026-01-22    32
2026-01-23    32
2026-01-24    29
2026-01-25    32
2026-01-26    32
2026-01-27    32
2026-01-28    32
2026-01-29    32
Name: count, dtype: int64


### Peak Day

In [155]:
peak_day = daily_volume.idxmax()
peak_volume = daily_volume.max()

print("Peak Day:", peak_day)
print("Number of Calls:", peak_volume)

Peak Day: 2026-01-10 00:00:00
Number of Calls: 70


# Team-level KPI Analysis

### KPI by Team

In [156]:
team_kpis = []

for team, group in df.groupby("queue"):

    total_calls = len(group)

    handled_calls = group["status"].isin(
        ["Answered", "Transferred"]
    ).sum()

    abandoned_calls = (
        group["status"] == "Abandoned"
    ).sum()

    transferred_calls = (
        group["status"] == "Transferred"
    ).sum()

    fcr_valid = group["first_contact_resolution"].dropna()

    team_kpis.append({
        "Team": team,
        "Volume": total_calls,
        "Answer Rate (%)": round(
            handled_calls / total_calls * 100, 2
        ),
        "Abandon Rate (%)": round(
            abandoned_calls / total_calls * 100, 2
        ),
        "AWT (sec)": round(
            group["wait_seconds"].mean(), 2
        ),
        "AHT (sec)": round(
            group["handle_seconds_clean"].mean(), 2
        ),
        "CSAT": round(
            group["csat_score"].mean(), 2
        ),
        "FCR (%)": round(
            (fcr_valid == "Yes").mean() * 100, 2
        ),
        "Transfer Rate (%)": round(
            transferred_calls / total_calls * 100, 2
        )
    })

team_kpis_df = pd.DataFrame(team_kpis)

team_kpis_df

,Team,Volume,Answer Rate (%),Abandon Rate (%),AWT (sec),AHT (sec),CSAT,FCR (%),Transfer Rate (%)
0,Billing,230,89.57,10.43,68.16,228.05,3.83,75.24,8.70
1,Sales,290,92.76,7.24,54.09,233.26,3.98,71.00,7.59
2,Support,444,82.66,17.34,142.50,406.21,3.18,72.21,14.19


### Call Reason Distribution by Team

In [157]:
team_reason = pd.crosstab(
    df["queue"],
    df["call_reason"]
)

team_reason

call_reason,Account Access,Billing Issue,Complaint,Order Status,Product Inquiry,Refund Request,Subscription Change,Technical Problem
queue,,,,,,,,
Billing,40,61,24,16,10,35,18,26
Sales,34,15,26,60,75,13,36,31
Support,83,32,49,39,39,29,23,150


### AHT by Call Reason

In [158]:
aht_by_reason = (
    df.groupby("call_reason")["handle_seconds_clean"]
      .mean()
      .round(2)
      .sort_values(ascending=False)
)

aht_by_reason

call_reason
Technical Problem      355.13
Complaint              322.76
Account Access         312.69
Order Status           297.58
Subscription Change    283.92
Billing Issue          280.75
Product Inquiry        279.95
Refund Request         276.38
Name: handle_seconds_clean, dtype: float64

### AHT by Team and Call Reason

In [159]:
aht_team_reason = (
    df.groupby(["queue", "call_reason"])["handle_seconds_clean"]
      .mean()
      .round(2)
      .unstack()
)

aht_team_reason

call_reason,Account Access,Billing Issue,Complaint,Order Status,Product Inquiry,Refund Request,Subscription Change,Technical Problem
queue,,,,,,,,
Billing,225.81,230.64,241.86,240.69,216.00,210.19,255.93,217.38
Sales,234.32,214.00,231.46,236.81,235.74,250.50,235.86,218.96
Support,395.63,405.43,413.23,415.85,406.52,397.80,381.64,412.84


### Hold Time, ACW and Transfer Rate by Team

In [160]:
team_operational = (
    df.groupby("queue")
      .agg(
          Hold_Time_sec=("hold_seconds", "mean"),
          ACW_sec=("after_call_work_seconds", "mean"),
          Transfer_Rate=(
              "status",
              lambda x: (x == "Transferred").mean() * 100
          )
      )
      .round(2)
)

team_operational

,Hold_Time_sec,ACW_sec,Transfer_Rate
queue,,,
Billing,42.13,48.11,8.70
Sales,43.74,47.42,7.59
Support,42.84,47.88,14.19


### AHT by Transfer Status

In [161]:
df["is_transferred"] = df["status"] == "Transferred"

aht_by_transfer = (
    df.groupby("is_transferred")["handle_seconds_clean"]
      .mean()
      .round(2)
)

aht_by_transfer

is_transferred
False    301.12
True     350.87
Name: handle_seconds_clean, dtype: float64

### Transfer Rate by Team and Call Reason

In [162]:
transfer_by_reason = (
    df.groupby(["queue", "call_reason"])["status"]
      .apply(lambda x: (x == "Transferred").mean() * 100)
      .round(2)
      .unstack()
)

transfer_by_reason

call_reason,Account Access,Billing Issue,Complaint,Order Status,Product Inquiry,Refund Request,Subscription Change,Technical Problem
queue,,,,,,,,
Billing,7.50,6.56,0.00,12.50,10.00,14.29,16.67,7.69
Sales,20.59,13.33,3.85,1.67,8.00,7.69,8.33,3.23
Support,16.87,3.12,12.24,12.82,20.51,10.34,13.04,15.33


### AHT by Team and Transfer Status

In [163]:
aht_transfer_team = (
    df.groupby(["queue", "is_transferred"])["handle_seconds_clean"]
      .mean()
      .round(2)
      .unstack()
)

aht_transfer_team

is_transferred,False,True
queue,,
Billing,227.17,236.30
Sales,232.42,242.64
Support,402.30,425.03


### Agent-level AHT Analysis

In [164]:
agent_aht = (
    df[df["queue"] == "Support"]
    .groupby("agent_id")
    .agg(
        Calls=("call_id", "count"),
        AHT_sec=("handle_seconds_clean", "mean"),
        Transfer_Rate=(
            "status",
            lambda x: (x == "Transferred").mean() * 100
        )
    )
    .round(2)
    .sort_values("AHT_sec", ascending=False)
)

agent_aht

,Calls,AHT_sec,Transfer_Rate
agent_id,,,
A07,29,526.62,20.69
A10,39,428.16,15.38
A02,40,407.22,20.00
A08,43,406.28,16.28
A12,42,400.52,11.90
A01,33,400.42,18.18
A03,35,399.77,5.71
A05,26,395.38,19.23
A04,37,387.81,5.41


### A07 AHT by Call Reason

In [165]:
df[
    (df["queue"] == "Support") &
    (df["agent_id"] == "A07")
].groupby("call_reason")["handle_seconds_clean"].agg(
    Calls="count",
    AHT_sec="mean"
).round(2)

,Calls,AHT_sec
call_reason,,
Account Access,2,510.00
Complaint,7,558.86
Order Status,3,499.67
Product Inquiry,3,527.00
Refund Request,2,472.00
Technical Problem,7,526.14


### A07 Operational Metrics

In [166]:
agent07_operational = (
    df[
        (df["queue"] == "Support") &
        (df["agent_id"] == "A07")
    ][
        ["handle_seconds_clean", "hold_seconds", "after_call_work_seconds"]
    ]
    .mean()
    .round(2)
)

agent07_operational

handle_seconds_clean       526.62
hold_seconds                42.25
after_call_work_seconds     51.79
dtype: float64

### Management Findings & Recommendations




| Finding | Evidence | Recommendation |
|---|---|---|
| Support در چند KPI عملیاتی مقادیر نامطلوب‌تری دارد | Answer Rate = 82.66%، Abandon Rate = 17.34%، AWT = 142.5s، AHT = 406.21s، CSAT = 3.18 | ظرفیت پاسخگویی و فرآیند رسیدگی در Support بررسی شود. |
| Call Mix به‌تنهایی AHT بالای Support را توضیح نمی‌دهد | AHT در Support برای تمام Call Reasonها بالاتر از دو تیم دیگر است | Workflow و نحوه رسیدگی به تماس‌ها در Support بررسی شود. |
| Transfer با AHT بالاتر همراه است، اما علت قطعی نیست | AHT تماس‌های Transferred = 350.87s در مقابل 301.12s برای سایر تماس‌ها | Routing و Knowledge Base برای Call Reasonهای دارای Transfer Rate بالا بررسی شوند. |
| بین Agentهای Support اختلاف AHT وجود دارد | A07 با 526.62s و 29 تماس، AHT بالایی دارد | Agentهای دارای AHT بالا با توجه به حجم تماس و Call Mix بررسی شوند. |